src/lib/zork/zork.ts

```diff
@@ -0,0 +1,378 @@
+/// <reference path='../../../third_party/wrtc/wrtc.d.ts' />
+
+/*
+ * Pure Node implementation of zork (no freedomjs).
+ * Rough lifecycle is to process single word commands such as "ping" until
+ * "give" or "get" is received, at which point a p2p proxy session is
+ * established and further input is treated as signaling channel messages.
+ * This logic can be found in `handleMsg()` below.
+ */
+
+
+// TODO: Use any specified `transform` settings once they are supported.
+//       Currently we stash them in the context but then ignore them.
+// TODO: Verify p2p proxy connection setup is sane:
+//       - dummy IGNORED datachannel necessary? (see "IGNORED" TODO below)
+//       - all required states or messages handled?
+//         (see "event.candidate -> TODO" below)
+//       - no resource leaks (see "cleanup" TODO below)
+// TODO: Implement backpressure (see "backpressure" TODO below)
+// TODO: Factor out logic duplicated in ../socks/bin/webrtc.ts
+// TODO: Replace console log calls with something more structured?
+// TODO: Do error states ever need to be indicated to clients rather than
+//       just logged locally?
+
+
+import * as constants from '../../generic_core/constants';
+import * as net from 'net';
+import * as node_server from '../socks/node/server';
+import * as node_socket from '../socks/node/socket';
+import * as socks_session from '../socks/session';
+import * as wrtc from 'wrtc';
+
+const RTC_PEER_CONFIG = {
+  iceServers: [
+    {url: 'stun:stun.l.google.com:19302'},
+    {url: 'stun:stun1.l.google.com:19302'},
+    {url: 'stun:stun.services.mozilla.com'}
+  ]
+};
+
+// The delimiter for Zork messages is just \n, but we use \r?\n for
+// MSG_DELIM_RE so that \r\n sent by e.g. a telnet client will also be treated
+// as a delimiter. https://en.wikipedia.org/wiki/Robustness_principle
+// "Be conservative in what you send, be liberal in what you accept."
+const MSG_DELIM_RE = /\r?\n/;
+
+const SOCKS_HOST = '0.0.0.0';
+const SOCKS_PORT = 9999;
+
+const ZORK_SERVER_PORT_START = 9000;
+const ZORK_SERVER_PORT_INCREMENT = 10;
+const ZORK_SERVER_BIND_MAXTRIES = 3;
+let zorkServerPort = ZORK_SERVER_PORT_START;
+let zorkServerBindNtries = 0;
+let nconnectionsMade = 0;
+let npeersGetting = 0;
+
+interface ParsedCmd {
+  id: string;        // e.g. 'ping', 'give', 'get', 'transform', etc.
+  source: string;    // e.g. 'transform with caesar'
+  tokens: string[];  // e.g. ['transform', 'with', 'caesar']
+}
+
+interface Context {
+  clientId: number;
+  socket: net.Socket;
+  mode: string;  // 'give', 'get', or null (still awaiting a 'give' or 'get' command)
+  peerConn: any; // using wrtc.RTCPeerConnection upsets tsc
```

Comment:
@trevj should I look into why tsc doesn't like a `wrtc.RTCPeerConnection` type declaration here?